File: ArchRefClient/ArchRefClient/src/app/shared/datamodel/topologymodel/relationshiptype.ts
import { Repository } from '../repository';
import { Property } from './property';

export class RelationshipType {

  id: number = null;
  name: string;
  repositoryRelationshipType: Repository;

  icon = '/assets/img/relationTypeDefault.png';
  providedProperties: Property[];

  constructor(name: string, repository: Repository) {
    this.name = name;
    this.repositoryRelationshipType = repository;
  }

  public getId(): number {
    return this.id;
  }

  public getName(): string {
    return this.name;
  }

  public getRepository(): Repository {
    return this.repositoryRelationshipType;
  }

  public setId(id: number) {
    this.id = id;
  }

  public setName(name: string) {
    this.name = name;
  }

  public setRepository(repositoryRelationshipType: Repository) {
    this.repositoryRelationshipType = repositoryRelationshipType;
  }

}

File: ArchRefServer/ArchRef_Server/src/main/java/de/arthurkaul/archref/model/topologyTemplate/RelationshipType.java
package de.arthurkaul.archref.model.topologyTemplate;

import java.util.Collection;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import de.arthurkaul.archref.model.Property;
import de.arthurkaul.archref.model.Repository;


@Entity
@Table(name="RELATIONSHIPTYPE")
public class RelationshipType{
	

	@Id
	@GeneratedValue()
	@Column(name="ID")
	private Long id;
	
	@Column(name="NAME")
	@NotNull
	private String name;
	
	@OneToMany (cascade=CascadeType.ALL, fetch=FetchType.LAZY, mappedBy="relationshipType")
	@JsonManagedReference (value="relationshipType-property")
	private Collection<Property> providedProperties;
	
	@ManyToOne(fetch=FetchType.LAZY)
	@JoinColumn(name="REPOSITORY_ID")
	@JsonBackReference
	private Repository repositoryRelationshipType;
	

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Repository getRepositoryRelationshipType() {
		return repositoryRelationshipType;
	}

	public void setRepositoryRelationshipType(Repository repositoryRelationshipType) {
		this.repositoryRelationshipType = repositoryRelationshipType;
	}

}
